### PR TITLE
MAINT: Remove deprecated binnings methods and args

### PR DESCRIPTION
**Pull request by MridulS**
Fixes https://github.com/scipp/scipp/issues/3409

### CODE REVIEW

File: docs/user-guide/binned-data/histogramming-grouping-and-binning.ipynb [hunk 1]
```diff
@@ -155,7 +155,7 @@
    "source": [
     "binned = table.bin(x=xbins)\n",
     "binned.coords['param'] = sc.array(dims=['x'], values=(np.random.random(39)*4).astype(np.int32))\n",
-    "grouped = binned.groupby('param').bins.concat('x')\n",
+    "grouped = binned.group('param')\n",
```
Comment (MridulS) [inline docs/user-guide/binned-data/histogramming-grouping-and-binning.ipynb:158]: I'm not too sure about this change here.

File: docs/user-guide/binned-data/histogramming-grouping-and-binning.ipynb [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "\n",
     "## Overview\n",
     "\n",
-    "Histogramming (see [sc.hist](../../generated/functions/scipp.hist.rst#scipp.hist)), grouping (using [sc.groupby](../groupby.ipynb)), and binning (see [Binned data](binned-data.ipynb)) all serve similar but slightly different purposes.\n",
+    "Histogramming (see [sc.hist](../../generated/functions/scipp.hist.rst#scipp.hist)), grouping (using [sc.group](../../generated/functions/scipp.group.html#scipp-group)), and binning (see [Binned data](binned-data.ipynb)) all serve similar but slightly different purposes.\n",
```
Comment (SimonHeybrock) [inline docs/user-guide/binned-data/histogramming-grouping-and-binning.ipynb:11]: `group` and `groupby` are not the same. `group` is a variant of `bin`. Also need to undo a bunch of changes below.

File: tests/binning_test.py [hunk 3]
```diff
@@ -586,14 +586,6 @@ def test_binning_low_level_functions_exist():
...
-    da = sc.data.table_xyz(100)
-    x = sc.linspace('x', 0, 1, num=10, unit='m')
-    with pytest.warns(UserWarning):
-        result = sc.histogram(da, bins=x)
-    assert sc.identical(result, da.hist(x=x))
-
-
 def test_bin_deprecated_arguments():
     da = sc.data.table_xyz(100)
     x = sc.linspace('x', 0, 1, num=10, unit='m')
```
Comment (SimonHeybrock) [inline tests/binning_test.py:591]: Isn't this one also being removed? Or does it fail for a different reason now?

Comment (MridulS) [reply tests/binning_test.py:591]: It was a ruff error, and this deprecation is not removed in this PR. I guess there is no reason to not remove it? This was also deprecated over 2 years ago.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**